extras/plot_throughput/plot_all.gp

```diff
@@ -0,0 +1,81 @@
+#!/usr/local/bin/gnuplot
+
+set tics font "arial,10"
+set terminal pdf size 6,4
+set key font "arial,10"
+
+set pointsize 0.5
+set bars 1.75
+
+array muarray[8]
+muarray[1] = 20; muarray[2] = 40; muarray[3] = 60; muarray[4] = 80; muarray[5] = 100; muarray[6] = 140; muarray[7] = 200; muarray[8] = 300;
+
+# CSV
+set datafile separator ','
+
+set xlabel "CPU Threads"
+set ylabel "Throughput (events/s)"
+do for [wrdid = 1:words(filenames)] {
+    wrd = word(filenames, wrdid)
+    input_file = 'clean/'.wrd
+    output_filename = 'plots/'.wrd[1:strlen(wrd)-4].".pdf"
+    set output output_filename
+    do for [i=1:8] {
+        set title 'mu'.muarray[i]
+        unset xrange
+        unset yrange
+        stats input_file using 1:i*2 nooutput
+        set xrange [0.7: 0.3+STATS_max_x]
+        set yrange [0: 1.1*STATS_max_y]
+        plot input_file using 1:i*2:i*2+1 title 'mu'.muarray[i] w yerr

```

Comment:
I agree this is quite fragile, but this happens all over this gnuplot script, the previous parts also assume each of the "clean" csv files have columns 17 columns corresponding to means and stddevs for all mu values, ordered in a specific way. 
The only way I see of getting around this would be by, when reading in each file, checking the headers of all the columns and retrieving their mu values. Then look up their corresponding stddev column by name as well. 
I'm not against this but it feels a bit beyond the purpose of gnuplot do so, and would rather assume the input data is super well-defined.

Reply:
Okay, that's totally cool with me.